examples/siro_sandbox/server/keyframe_utils.py

```diff
@@ -57,20 +64,29 @@ def update_consolidated_keyframe(consolidated_keyframe, inc_keyframe):
     # for a deletion, just remove all references to this instanceKey
     if "deletions" in inc_keyframe:
         inc_deletions = inc_keyframe["deletions"]
+        for key in inc_deletions:
+            # If we find a corresponding creation in the con keyframe, we can remove
+            # the creation and otherwise skip this deletion. This logic ensures
+            # consolidated keyframes don't get bloated as many items are added
+            # and removed over time.
+            con_creations = consolidated_keyframe["creations"]
+            for entry in con_creations:
+                if entry["instanceKey"] == key:
+                    con_creations.remove(entry)
+                    found = True
+                    break
+            if not found:
+                # if we didn't find the creation, then we should still include the deletion
+                ensure_list(consolidated_keyframe, "deletions")
```

Comment:
This is essential new logic to ensure this function can consolidate multiple incremental keyframes into one. (Previously, I was only every consolidating into an "everything since the start of time" keyframe.)